Validation/RecoTrack/python/PostProcessorTracker_cfi.py

```diff
@@ -313,6 +316,27 @@ def _addNoFlow(module):
     postProcessorTrackSummary
 )
 
+fastSim.toModify(postProcessorTrack, subDirs = [e for e in _defaultSubdirs if e not in ["Tracking/TrackGsf/*","Tracking/TrackConversion/*"]])
+fastSim.toModify(postProcessorTrackSummary, subDirs = [e for e in _defaultSubdirs if e not in ["Tracking/TrackGsf/*","Tracking/TrackConversion/*"]])
+
+#######
+# Define a standalone seuquence to support the Standalone harvesting mode
+# see https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideMultiTrackValidator#cmsDriver_MTV_alone_i_e_standalone for more information
+########
+
+postProcessorTrackStandalone = postProcessorTrack.clone(
+    subDirs = _defaultSubdirs.append("Tracking/TrackBHadron/*"),
```

Comment:
What is the intention of this line

 If I read the code correctly, right now it appends `"Tracking/TrackBHadron/*"` to `_defaultSubdirs` and assigns `None` to  `subDirs`, which means the `subDirs` parameter will get deleted for `postProcessorTrackStandalone`.

Reply:
the intention is to append `"Tracking/TrackBHadron/*"` to `_defaultSubdirs`.
I have not tested the behaviour in standalone mode, as there is no driver command for that in the matrix.
Will:
```diff
diff --git a/Validation/RecoTrack/python/PostProcessorTracker_cfi.py b/Validation/RecoTrack/python/PostProcessorTracker_cfi.py
index 9f5f0c00f38..965b67abd16 100644
--- a/Validation/RecoTrack/python/PostProcessorTracker_cfi.py
+++ b/Validation/RecoTrack/python/PostProcessorTracker_cfi.py
@@ -325,10 +325,10 @@ fastSim.toModify(postProcessorTrackSummary, subDirs = [e for e in _defaultSubdir
 ########
 
 postProcessorTrackStandalone = postProcessorTrack.clone(
-    subDirs = _defaultSubdirs.append("Tracking/TrackBHadron/*"),
+    subDirs = _defaultSubdirs+["Tracking/TrackBHadron/*"]
 )
 postProcessorTrackSummaryStandalone = postProcessorTrackSummary.clone(
-    subDirs = _defaultSubdirs.append("Tracking/TrackBHadron/*"),
+    subDirs = _defaultSubdirs+["Tracking/TrackBHadron/*"]
 )
```
be OK?
I've tested now with:
```
cmsDriver.py step4 --conditions auto:phase1_2017_realistic -s HARVESTING:postProcessorTrackSequenceStandalone --filetype DQM --filein file:step3_inDQM.root --no_exec
```
and it does the trick.
Let me know if there is a more pythonic way of dealing with it...

Reply:
> Will:
> 
> ```diff
> diff --git a/Validation/RecoTrack/python/PostProcessorTracker_cfi.py b/Validation/RecoTrack/python/PostProcessorTracker_cfi.py
> index 9f5f0c00f38..965b67abd16 100644
> --- a/Validation/RecoTrack/python/PostProcessorTracker_cfi.py
> +++ b/Validation/RecoTrack/python/PostProcessorTracker_cfi.py
> @@ -325,10 +325,10 @@ fastSim.toModify(postProcessorTrackSummary, subDirs = [e for e in _defaultSubdir
>  ########
>  
>  postProcessorTrackStandalone = postProcessorTrack.clone(
> -    subDirs = _defaultSubdirs.append("Tracking/TrackBHadron/*"),
> +    subDirs = _defaultSubdirs+["Tracking/TrackBHadron/*"]
>  )
>  postProcessorTrackSummaryStandalone = postProcessorTrackSummary.clone(
> -    subDirs = _defaultSubdirs.append("Tracking/TrackBHadron/*"),
> +    subDirs = _defaultSubdirs+["Tracking/TrackBHadron/*"]
>  )
> ```
> 
> be OK?

Yes, this would be the best way (that I am aware of) to use `defaultSubdirs` and `"Tracking/TrackBHadron/*"` for `subDirs`.

Reply:
thanks, implemented like that now in  eb6dd4c